diff -u a/config.def.h b/config.def.h
--- a/config.def.h	2021-02-10 08:24:52.406603894 +0100
+++ b/config.def.h	2021-02-10 08:25:03.569937015 +0100
@@ -6,7 +6,7 @@
 static int min_width = 500;                    /* minimum width when centered */
 /* -fn option overrides fonts[0]; default X11 font or font set */
 static const char *fonts[] = {
-	"monospace:size=10"
+	"SauceCodePro Nerd Font:style=Regular:size=11"
 };
 static const char *prompt      = NULL;      /* -p  option; prompt to the left of input field */
 static const char *colors[SchemeLast][2] = {
