diff -u a/config.def.h b/config.def.h
--- a/config.def.h	2021-02-07 15:26:57.453208816 +0100
+++ b/config.def.h	2021-02-07 15:27:52.163208460 +0100
@@ -73,7 +73,7 @@
 /* commands */
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
 static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
-static const char *termcmd[]  = { "st", NULL };
+static const char *termcmd[]  = { "tabbed", "-r", "2", "-c", "st", "-w", "''", NULL };
 
 static Key keys[] = {
 	/* modifier                     key        function        argument */
